client/tests/gui/source/delete/test_dialog.py

```diff
@@ -1,22 +1,91 @@
-import unittest
+import pytest
 
-from securedrop_client.gui.source.delete import DeleteSourceDialog as Dialog
+from securedrop_client.gui.source.delete import DeleteSourceDialog
 from tests import factory
 
 
-class DeleteSourceDialogTest(unittest.TestCase):
-    def setUp(self):
-        self._source = factory.Source()
-        factory.File(source=self._source)
-        self.dialog = Dialog(self._source)
+@pytest.fixture(
+    params=[set(), set([factory.Source()]), set([factory.Source(), factory.Source()])],
+)
+def dialog(request):
+    """
+    Set up the dialog under various conditions: 0 sources, 1 source, >1 source.
+    Tests that target a specific configuration can use decorators to skip unwanted
+    conditions.
+    """
+    # Give the source(s) a submission
+    for source in request.param:
+        factory.File(source=source)
+    return DeleteSourceDialog(request.param)
 
-    def test_default_button_is_safer_choice(self):
+
+class TestDeleteSourceDialog:
```

Comment:
I appreciate the switch from unittest to pytest. Now that dialog is a fixture and not a member variable, there's no need for a class since `self` is never used AFAICT unless you find it useful for organization (I personally don't).

Reply:
I do honestly like it for organization and consistency with other unittests (and for ease of doing things like parameterization on the whole test class if wanted) and it's a style I default to, but ... I am open to other thoughts :)